config/typescript/tsconfig.base.json

```diff
@@ -1,33 +1,14 @@
 {
-  "compileOnSave": false,
+  "extends": "@shopify/typescript-configs/library.json",
   "compilerOptions": {
     "composite": true,
-    "declaration": true,
-    "declarationMap": true,
-    "pretty": true,
-    "moduleResolution": "node",
-    "target": "es5",
-    "downlevelIteration": true,
-    "esModuleInterop": true,
-    "jsx": "react",
-    "isolatedModules": true,
-    "strictNullChecks": true,
+    "__TODOCOMMENT1": "Strict mode enables these strict* and noImplicit* options.",
```

Comment:
Interesting, what are these?

Reply:
Comments are valid in tsconfig files, but @atesgoral asked about keeping valid json; https://github.com/Shopify/quilt/pull/1811#discussion_r605777199 


So I made up some keys that do nothing.

Personally I'm not a fan and would prefer actual comments but it's not a hill i'm gonna die on

Reply:
Ah well, turns out arbitary strings cause type check to fail. so back to real comments.